MonoDevelop.MSBuild.Editor/Completion/MSBuildCompletionSource.cs

```diff
@@ -119,19 +125,141 @@ CompletionContext CreateCompletionContext (List<CompletionItem> items)
 
 			foreach (var att in rr.GetAttributeCompletions (doc, doc.ToolsVersion)) {
 				if (!existingAtts.ContainsKey (att.Name)) {
-					items.Add (CreateCompletionItem (att));
+					items.Add (CreateCompletionItem (att, MSBuildSpecialCommitKind.Attribute));
 				}
 			}
 
 			return CreateCompletionContext (items);
 		}
 
-		CompletionItem CreateCompletionItem (BaseInfo info, string prefix = null)
+		//Dictionary<int, string> cache = new Dictionary<int, string> ();
+
+		protected override async Task<CompletionContext> GetAttributeValueCompletionsAsync (
+			IAsyncCompletionSession session,
+			SnapshotPoint triggerLocation,
+			List<XObject> nodePath,
+			IAttributedXObject attributedObject,
+			XAttribute attribute,
+			CancellationToken token)
+		{
+			/*if (!attribute.NameEquals ("include", true)) {
+				return CompletionContext.Empty;
+			}*/
+
+			var context = await GetSessionContext (session, triggerLocation, token);
+			var rr = context.rr;
+			var doc = context.doc;
+
+			if (rr == null) {
+				return CompletionContext.Empty;
+			}
+
+			if (attributedObject is XElement xElement &&
+				xElement.NameEquals ("PackageReference", true)) {
```

Comment:
Do you need to consider casing in this comparison?

Reply:
No longer using it. But casing could have been removed.